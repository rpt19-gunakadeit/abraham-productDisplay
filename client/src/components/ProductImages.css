.pageTitle {
  text-align: center;
}

.navbar {
  width: 100%;
  max-width: 100%;
}

img.defaultSize {
  display: inline-block;
  width: 463px;
  height: 578px;
  padding-left: 12px;
  padding-bottom: 4px;
}
/*
#modal, .img-modal {
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 90vw;
  max-width: 100%;
  height: 100vh;
  max-height: 100%;
  z-index: 100;
  background-color: rgb(0,0,0,0.5);
} */

h1, h2 {
  color: black;
}

img.modal-content {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  padding: 20px 50px 20px 20px;
  overflow: auto;
}

img.modalImg {
  position: fixed;
  display: block;
  width: 550px;
  height: 600px;
  top: 50%;
  left: 50%;
}

/* .text {
  color: white;
  size: 36px;
} */

.modal-images {
  position: fixed;
  /* background-color: rgb(0,0,0,0.4); */
}

span.close {
  color: white;
  padding-left: 95%;
}

.display-block {
  display: block;
}

.display-none {
  display: none;
}

/* */