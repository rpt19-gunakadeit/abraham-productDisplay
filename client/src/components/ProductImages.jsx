import React from 'react';
import $ from 'jquery';

class ProductImages extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      defaultImages: [],
      largeImages: []
    }
  }

  componentDidMount() {
    this.displayImages();
  }

  displayImages() {
    let params = new URL(window.location.href)
    let styleId = params.searchParams.get('');
    console.log('client styleId: ', styleId)
    $.ajax({
      //retrieve medium sized images
      url: `http://localhost:3000/:${styleId}`,
      method: 'GET',
      dataType: 'json',
      success: (data) => {
        console.log('data: ', data)
        this.setState({
          defaultImages: data
        })
      },
      error: (err) => {
        console.log('Error: ', err);
      }
    })
  }

  render() {
    let images = this.state.defaultImages;
    return (
      <div className="defaultDisplay">
        <h1 className="pageTitle">Nike</h1>
        {images.map(img => (
          <img
          className="defaultSize"
          key={img.id}
          src={img.mediumUrl}
          />
        ))}
      </div>
    )
  }
}

export default ProductImages;