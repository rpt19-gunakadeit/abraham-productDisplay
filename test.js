import React from 'react';
import ProductImages from './client/src/components/ProductImages.jsx';
import renderer from 'react-test-renderer';
import {mount, shallow} from 'enzyme';

test('Correctly render <ProductImages/>', () => {
  const tree = renderer.create(<ProductImages/>).toJSON();
  expect(tree).toMatchSnapshot();
})