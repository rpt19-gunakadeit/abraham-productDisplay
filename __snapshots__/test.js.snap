// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly render <ProductImages/> 1`] = `
<div
  className="defaultDisplay"
>
  <h1
    className="pageTitle"
  >
    Nike
  </h1>
</div>
`;
